File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        arr = [0, 1, 1]
        
        if n > 2:
            for i in range(3, n+1):
                arr.append(arr[i-3] + arr[i-2] + arr[i-1])
        
        return arr[n]

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.swift
class Solution {
    func tribonacci(_ n: Int) -> Int {
        var result = [0, 1, 1]
        
        if n > 2 {
            for i in 3...n {
                result.append(result[i-3] + result[i-2] + result[i-1])
            }
        }
        
        return result[n]
    }
}